src/platforms/rp2040/src/lib/networkdriver.c

```diff
@@ -236,7 +259,54 @@ static term start_sta(term sta_config, GlobalContext *global)
         }
     }
 
-    cyw43_arch_enable_sta_mode();
+    if (term_is_invalid_term(hostname_term)) {
+        driver_data->hostname = malloc(DEFAULT_HOSTNAME_SIZE);
+        if (IS_NULL_PTR(driver_data->hostname)) {
+            free(ssid);
+            free(psk);
+            cyw43_arch_disable_sta_mode();
+            return OUT_OF_MEMORY_ATOM;
+        } else {
+            term error = get_default_device_name(driver_data->hostname, global);
+            if (error != OK_ATOM) {
+                free(ssid);
+                free(psk);
+                free(driver_data->hostname);
+                cyw43_arch_disable_sta_mode();
+                return error;
+            }
+        }
+    } else {
+        int ok = 0;
+        char *buf = interop_term_to_string(hostname_term, &ok);
+        if (!ok || IS_NULL_PTR(buf)) {
+            free(ssid);
+            free(psk);
+            if (buf != NULL) {
+                free(buf);
+                cyw43_arch_disable_sta_mode();
+                return BADARG_ATOM;
+            }
+            cyw43_arch_disable_sta_mode();
+            return OUT_OF_MEMORY_ATOM;
+        }
+        size_t buf_size = strlen(buf);
+        driver_data->hostname = malloc(buf_size);
+        if (IS_NULL_PTR(driver_data->hostname)) {
+            free(ssid);
+            free(psk);
+            free(buf);
+            cyw43_arch_disable_sta_mode();
+            return OUT_OF_MEMORY_ATOM;
+        }
+        memcpy(driver_data->hostname, buf, buf_size);
```

Comment:
strlen + malloc + memcpy = strdup ;)
In general: let's just use strdup!
In this specific case I didn't understand why we don't just keep the buffer return from `interop_term_to_string`.